### PR TITLE
fix(chat): publication mobile change overlap (Issue #1620)

### PR DESCRIPTION
**Pull request by Derikyan**
**Description:**

Change path form should be open on click on Change link only, not to any place inside text box
Issue also should be fixed for select folders in Manage Attachments
Expected result: Open select folders form on click Change link only, not to any area in text box

Issues:

- Issue #1620 

**UI changes**

Снимок экрана 2024-06-20 в 13 55 37

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1627.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/Publish/PublishWizard.tsx [hunk 1]
```diff
@@ -180,8 +181,19 @@ export function PublishModal({
     dispatch(PublicationActions.uploadRules({ path }));
   }, [dispatch, path]);
 
-  const handleFolderChange = useCallback(() => {
-    setIsChangeFolderModalOpened(true);
+  const handleFolderChangeForMobile = useCallback<
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/Publish/PublishWizard.tsx]: Do we really need 2 functions? Why we can't just move onClick handler to `span`  with `change` text

File: apps/chat/src/components/Files/PreUploadModal.tsx [hunk 2]
```diff
@@ -305,8 +306,19 @@ export const PreUploadDialog = ({
     [folderPath, selectedFiles],
   );
 
-  const handleFolderChange = useCallback(() => {
-    setIsChangeFolderModalOpened(true);
+  const handleFolderChangeForMobile = useCallback<
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Files/PreUploadModal.tsx]: same

**Comment by Derikyan** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/488212)

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1627.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1627.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1627.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**